### PR TITLE
feat: Adds Byond detection support

### PR DESCRIPTION
**Pull request by CorruptComputer**
I finally got around to this, but this is to support this PR: https://github.com/sourcerer-io/awesome-libraries/pull/104

Please let me know if any changes are needed!

### CODE REVIEW

**Comment by yaronskaya** [comment]
Hi @CorruptComputer. Thank you for adding that. 
Is Byond a part of DM language? Github classify it as DM language now.
What is a way to import libraries in code in that language?

**Comment by CorruptComputer** [comment]
Hi @yaronskaya DM is the custom language Byond created for their game engine. I don't believe that libraries are able to be imported, as all of the code written must be attached to an object in the game world.

**Comment by yaronskaya** [comment]
@CorruptComputer Would it be more correct to assign DM as language of a code? 
Regarding the libraries, what are these items that you provided in [awesome-libraries PR](https://github.com/sourcerer-io/awesome-libraries/pull/104)? Are these libraries or not?

I'm asking because we need to clarify the language specifics before reviewing PRs.

**Comment by CorruptComputer** [comment]
Possibly? Those are the different kinds of files you can have in the project, for example `.dm` is for the code. `.dmm` is for the mapping, and `.dmi` is for the sprites you create within the editor. 

The only reason I lumped them all into `Byond` is that none of the code can be used outside of this engine.

**Comment by yaronskaya** [comment]
@CorruptComputer got it. 
1. Let's assign .dm, .dmi, .dmm to DM language in Heuristics.kt
2. Let's assign .dmi, .dmm to Byond technology. See [example](https://github.com/sourcerer-io/sourcerer-app/blob/d055504cd82a3b3af2b5be3432973fcfa60a1ea3/src/main/kotlin/app/extractors/JavascriptExtractor.kt#L27) where we assigned all .vue files to Vue technology in javascript. Basically, you will need to create DMExtractor and override extractLibStats function. 
3. I'll add the combining of DM files and Byond technology to Byond on our backend so that when a repo has Byond technology and .dm files then .dm will be threated as Byond.

Does that work for you?

**Comment by CorruptComputer** [comment]
That works for me :+1:

I'll start working on that now.

**Comment by CorruptComputer** [comment]
I believe I have made the requested changes, I'm not very familiar with Kotlin and this is actually my first time working with it. Please tell me if anything else needs changes!

File: src/main/kotlin/app/extractors/Heuristics.kt [hunk 1]
```diff
@@ -450,6 +450,15 @@ val HeuristicsMap = mapOf<String, (String, String) -> ExtractorInterface?>(
...
         CommonExtractor(Lang.IDL)
     },
+    "dm" to { _, _ ->
+        CommonExtractor(Lang.DM)
+    },
+    "dmm" to { _, _ ->
+        CommonExtractor(Lang.DM)
+    },
+    "dmi" to { _, _ ->
+        CommonExtractor(Lang.DM)
```
Comment (yaronskaya) [inline src/main/kotlin/app/extractors/Heuristics.kt]: Replace with `DMExtractor()`

File: src/main/kotlin/app/extractors/DMExtractor.kt [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            type = ExtractorInterface.TYPE_LANGUAGE,
+            tech = "byond.mapping"
+        )).filter { it.numLinesAdded > 0 || it.numLinesDeleted > 0 }
+
+        // Add stats from *.dmi files.
+        val spriteStats = listOf(CommitStats(
+            numLinesAdded = spriteFiles.map { it.getAllAdded().size }.sum(),
+            numLinesDeleted = spriteFiles.map { it.getAllDeleted().size }.sum(),
+            type = ExtractorInterface.TYPE_LANGUAGE,
+            tech = "byond.sprites"
```
Comment (yaronskaya) [inline src/main/kotlin/app/extractors/DMExtractor.kt]: Replace with `ExtractorInterface.TYPE_LIBRARY`, `"dm.byond-sprites"`

File: src/main/kotlin/app/extractors/DMExtractor.kt [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            type = ExtractorInterface.TYPE_LANGUAGE,
+            tech = "byond.dm"
+        )).filter { it.numLinesAdded > 0 || it.numLinesDeleted > 0 }
+
+        // Add stats from *.dmm files.
+        val mapStats = listOf(CommitStats(
+            numLinesAdded = mapFiles.map { it.getAllAdded().size }.sum(),
+            numLinesDeleted = mapFiles.map { it.getAllDeleted().size }.sum(),
+            type = ExtractorInterface.TYPE_LANGUAGE,
+            tech = "byond.mapping"
```
Comment (yaronskaya) [inline src/main/kotlin/app/extractors/DMExtractor.kt]: Replace with `ExtractorInterface.TYPE_LIBRARY`, `"dm.byond-mapping"`

File: src/main/kotlin/app/extractors/DMExtractor.kt [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        val mapExtension = ".dmm"
+        val spriteExtension = ".dmi"
+
+        val codeFiles = files.filter { it.path.endsWith(codeExtension) }
+        val mapFiles = files.filter { it.path.endsWith(mapExtension) }
+        val spriteFiles = files.filter { it.path.endsWith(spriteExtension) }
+        
+
+        // Add stats from *.dm files.
+        val codeStats = listOf(CommitStats(
```
Comment (yaronskaya) [inline src/main/kotlin/app/extractors/DMExtractor.kt]: Do not process *.dm in this section. Mapping to byond technology will be added on server.

**Comment by CorruptComputer** [comment]
@yaronskaya I've made the requested changes, are there any changes needing to be made to sourcerer-io/awesome-libraries#104?

**Pull request merged**